### PR TITLE
Publish: Cascading version update

### PR DESCRIPTION
**Pull request by MariusStorhaug**
# Change

Main functionality changes:
- Added ability to do cascading updates to "latest" versions for major and major.minor for Template Specs and Bicep Registry. This will allow consumers to specify a major, or major.minor version in the template reference. i.e.:
  - `br:acr.azurecr.io/bicep/modules/microsoft.storage.storageaccount:1` to always get the latest non-breaking changes (features and patches/fixes) to their deployment,
  - `br:acr.azurecr.io/bicep/modules/microsoft.storage.storageaccount:1.2` to only get the latest patches for their module reference, this in addition to the already able way with
  - `br:acr.azurecr.io/bicep/modules/microsoft.storage.storageaccount:1.2.123` for a specific version.

`Get-ModulesToUpdate`
- Rename script and functions from `*Update` -> `*Publish` to align better with its usage.
- Reduce verbose logging where this was more noise than helpful

Pipelines:

- [Storage Account | ADO](https://dev.azure.com/servicescode/infra-as-code-source/_build/results?buildId=195786&view=results)
- [Build Status](https://dev.azure.com/servicescode/infra-as-code-source/_build/latest?definitionId=2232&branchName=refs%2Fpull%2F978%2Fmerge)
- [Storage: StorageAccounts](https://github.com/Azure/ResourceModules/actions/workflows/ms.storage.storageaccounts.yml)

### Know issues:
Running the pipelines in ADO for a given module multiple times on the same commit will fail for the publishing step as it will attempt to publish a package using the same version which is not supported for ADO Universal Packages.

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update (Wiki)

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [x] I did format my code

### CODE REVIEW

File: .azuredevops/modulePipelines/ms.analysisservices.servers.yml [hunk 1]
```diff
@@ -6,7 +6,11 @@ parameters:
     type: boolean
     default: true
   - name: prerelease
-    displayName: Publish prerelease module
+    displayName: Publish prerelease
+    type: boolean
+    default: false
+  - name: publishAll
+    displayName: Publish module and all child modules
```
Comment (MrMCake) [inline .azuredevops/modulePipelines/ms.analysisservices.servers.yml:13]:
```suggestion
    displayName: Force publish module and all child modules
```

Let's have a word on that in the group. For now just a small adjustment proposal ;)

File: utilities/pipelines/resourcePublish/Get-ModulesToPublish.ps1 [hunk 2]
```diff
@@ -31,13 +31,13 @@ function Get-ModifiedFileList {
...
-    Write-Verbose "Gathering modified files between [$Commit] and [$CompareCommit]" -Verbose
+    Write-Verbose "Gathering modified files between [$CompareCommit] and [$Commit]" -Verbose
     $Diff = git diff --name-only --diff-filter=AM $CompareCommit $Commit
-    $ModifiedFiles = $Diff | Get-Item
-    Write-Verbose 'The following files have been added or modified:' -Verbose
-    $ModifiedFiles | ForEach-Object {
-        Write-Verbose (' - [{0}]' -f $_.FullName) -Verbose
-    }
+    $ModifiedFiles = $Diff | Get-Item -Force
+    # Write-Verbose 'The following files have been added or modified:' -Verbose
```
Comment (MrMCake) [inline utilities/pipelines/resourcePublish/Get-ModulesToPublish.ps1:37]: Remove comments

File: utilities/pipelines/resourcePublish/Get-ModulesToPublish.ps1 [hunk 3]
```diff
@@ -62,22 +62,22 @@ function Get-GitBranchName {
     param ()
 
     # Get branch name from Git
-    try {
-        Write-Verbose "Git: Using git command 'git branch --show-current'" -Verbose
-        $BranchName = git branch --show-current
-    } catch {
-        Write-Verbose 'Git: No name found.' -Verbose
-    }
+    #try {
```
Comment (MrMCake) [inline utilities/pipelines/resourcePublish/Get-ModulesToPublish.ps1:65]: Remove comments

File: .azuredevops/modulePipelines/ms.analysisservices.servers.yml (hunk 1 above)
Comment (MariusStorhaug) [reply .azuredevops/modulePipelines/ms.analysisservices.servers.yml:13]: This makes more sense, though I removed the "Force" wording based on initial feedback. In retrospect, that might have been based on the usage of `force` in powershell.

File: utilities/pipelines/resourcePublish/Get-ModulesToPublish.ps1 (hunk 2 above)
Comment (MariusStorhaug) [reply utilities/pipelines/resourcePublish/Get-ModulesToPublish.ps1:37]: Will remove comments, to reduce the verbosity of the logging as discussed earlier.

Comment (MariusStorhaug) [reply utilities/pipelines/resourcePublish/Get-ModulesToPublish.ps1]: Done

File: .azuredevops/modulePipelines/ms.analysisservices.servers.yml (hunk 1 above)
Comment (MrMCake) [reply .azuredevops/modulePipelines/ms.analysisservices.servers.yml:13]: Agree. I was trying to find a different wording, but couldn't find one. :D

Comment (MariusStorhaug) [reply .azuredevops/modulePipelines/ms.analysisservices.servers.yml]: We opted to remove this feature, so closing this topic.

**Comment by MariusStorhaug** [comment]
Ready for review

**Review by MrMCake — APPROVED** [review]

**Pull request merged**